### PR TITLE
upgrade to Documenter 0.20

### PR DESCRIPTION
**Pull request by visr**
replaces #73

I'm stuck however, not sure what's wrong, but it just stops building the docs halfway without an error message:
```
λ julia --color=yes make.jl
Documenter: setting up build directory.
Documenter: expanding markdown templates.
```

### CODE REVIEW

**Comment by visr** [comment]
Ah it is because the quickstart example is crashing julia.
```julia
julia> nccreate(fn,"rad","lon",lon,lonatts,"lat",lat,latatts,"time",tim,timatts,atts=varatts);

Unreachable reached at 000000001CBDF06C

Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ILLEGAL_INSTRUCTION at 0x1cbdf06c -- #NcVar#6 at C:\Users\visser_mn\.julia\dev\NetCDF\src\NetCDF.jl:143
in expression starting at no file:0
#NcVar#6 at C:\Users\visser_mn\.julia\dev\NetCDF\src\NetCDF.jl:143
unknown function (ip: 000000001CBDF0E8)
```

**Comment by visr** [comment]
Above issue is fixed by #78. Still needs some fixing up though, for the examples in the docs.

**Comment by visr** [comment]
@fredrikekre since you kindly offered help in https://discourse.julialang.org/t/psa-use-a-project-for-building-your-docs/14974

How can it be that the docs are failing with this:
```
$ julia --project=docs/ docs/make.jl
ERROR: LoadError: LoadError: libnetcdf not properly installed. Please run Pkg.build("NetCDF")
```

Even though the tests are working just fine. Do we need to explicitly call `Pkg.build("NetCDF")`? I'm also a bit confused, in some `docs/Project.toml` the package itself is listed, in others not:

https://github.com/JuliaGraphics/Colors.jl/blob/835d4454e36b34fcec4bd5255779202ca2b5a358/docs/Project.toml
https://github.com/fredrikekre/Literate.jl/blob/99e74daf79da4fb29e05ea40588416d096a21eec/docs/Project.toml

Or is it needed to list NetCDF if you use it in building the docs (like we do)? Wouldn't this add the latest release though?

**Comment by fredrikekre** [comment]
Yea, `Pkg.develop` does not result in a call to build, so if you have binary dependencies you will also need to add `Pkg.build("NetCDF")` after the call to `Pkg.develop`, see  https://github.com/JuliaLang/Pkg.jl/issues/862 and the link to the discourse conversation in there.

It is fine to include the package itself in the Project.toml file, but not needed since the call to `Pkg.develop` will add it, and make sure you are not using the released version of the package.

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,13 @@ notifications:
...
-  - julia -e 'using Pkg; Pkg.add("Documenter")'
-  - julia -e 'using Pkg; Pkg.dir("NetCDF"); include(joinpath("docs", "make.jl"))'
+jobs:
+  include:
+    - stage: "Documentation"
+      julia: 1.0
+      os: linux
+      script:
+        - julia --project=docs/ -e 'using Pkg; Pkg.instantiate();
+                                    Pkg.develop(PackageSpec(path=pwd()))'
```
Comment (fredrikekre) [inline .travis.yml]:
```suggestion
                                    Pkg.develop(PackageSpec(path=pwd())); Pkg.build("NetCDF")'
```

**Comment by visr** [comment]
Ah that makes sense, many thanks!

**Comment by fredrikekre** [comment]
:tada:

**Pull request merged**